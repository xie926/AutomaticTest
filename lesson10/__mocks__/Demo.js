export const fetchData = () => {
  return new Promise((resolved, rejected) => {
    resolved("(function(){return '123'})()")
    // resolved({
    //   data: "(function(){return '123'})()"
    // })
  })
}