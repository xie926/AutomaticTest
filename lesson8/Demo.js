import axios from 'axios'
export const runCallBack = (callBack) => {
  callBack()
}

export const runCallBack2 = (callBack) => {
  callBack('abd')
}

export const createObject = (classItem) => {
  new classItem();
}

export const getData = () => {
  return axios.get('./api').then(res => res.data)
}

export const forEach = (items, callback) => {
  for (let index = 0; index < items.length; index++) {
    callback(items[index]);
  }
}