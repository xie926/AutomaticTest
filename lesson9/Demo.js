export const generateConfig = () => {
  return {
    server: 'http://localhost',
    port: 8000,
    domain: 'localhost',
    time: 20191,
    date: new Date()
  }
}

export const generateAnthorConfig = () => {
  return {
    server: 'http://localhost',
    port: 8000,
    domain: 'localhost',
    time: 20191
  }
}