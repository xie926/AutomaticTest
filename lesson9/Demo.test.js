import { generateConfig, generateAnthorConfig } from "./Demo";

// test('测试 generateConfig 函数', () => {
//   expect(generateConfig()).toEqual({
//     server: 'http://localhost',
//     port: 8000,
//     domain: 'localhost' // 当我们修改测试用例的时候，导致这个测试不通过，然后就得去修改原有的配置文件，很麻烦，由此引出snapshot
//   })
// });

test("测试 generateConfig 函数", () => {
  expect(generateConfig()).toMatchSnapshot({
    date: expect.any(Date)
  });
});

test("测试 generateAnotherConfig 函数", () => {
  expect(generateAnthorConfig()).toMatchInlineSnapshot(`
    Object {
      "domain": "localhost",
      "port": 8000,
      "server": "http://localhost",
      "time": 20191,
    }
  `);
});
