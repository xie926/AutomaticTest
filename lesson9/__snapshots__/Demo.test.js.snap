// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateConfig 函数 1`] = `
Object {
  "date": Any<Date>,
  "domain": "localhost",
  "port": 8000,
  "server": "http://localhost",
  "time": 20191,
}
`;
